### PR TITLE
Inline block support

### LINKED ISSUE #2601: Add support for inline block to support multiple languages
**Is your feature request related to a problem? Please describe.**
We currently do not support the block with `type=inline`. The article https://www.test.bbc.co.uk/news/articles/cv7yn5n25rno is an English article containing a Farsi block.

**Describe the solution you'd like**
Give the payload https://www.test.bbc.co.uk/news/articles/cv7yn5n25rno we should render the inline block. 

We can also look to add the `dir` and `lang` attribute to the HTML elements that map to the given `language` of the block.

The following requirements must be satisfied (originally from https://github.com/bbc/simorgh/issues/1912)

As a screen-reader user
I want text to be read out in the appropriate voice for its language
So that I can understand it

Given I am on an article
When I reach a fragment with a language attribute that is different from the language attribute of the article
Then I hear it read in the appropriate screen-reader voice

**Describe alternatives you've considered**
N/A

**Testing notes**
[Tester to complete]

Dev insight: Snapshot tests are required and manual regression of the article. 

**Additional context**
We should add fixture data matching the example - https://www.test.bbc.co.uk/news/articles/cv7yn5n25rno

### PR DESCRIPTION
**Pull request by twinlensreflex**
Resolves #2601

**Overall change:** 
Adds support for the `Inline` block.

**Code changes:**

- Adds `Inline` component which is a span with `lang` attribute
- Adds new inline componnt to `componentsToRender` in Paragraph
- Adds tests

**Testing notes:**
* Data with inline blocks for testing is in `71ca84e` (reverted on this branch)

```
$ git checkout 71ca84 data/news/articles/cv7yn5n25rno.json
```
Go to http://localhost.bbc.com:7080/news/articles/cv7yn5n25rno

---

- [ ] I have assigned myself to this PR and the corresponding issues
- [ ] Tests added for new features
- [ ] Test engineer approval

### CODE REVIEW

File: src/app/containers/InlineContainer/index.jsx [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+
+const componentsToRender = { fragment, urlLink: InlineLink };
+
+const InlineContainer = ({ blocks, language }) => {
+  const { script, service } = useContext(ServiceContext);
+
+  return (
+    <GridItemConstrainedMedium>
+      <InlineSpan lang={language} script={script} service={service}>
+        <Blocks blocks={blocks} componentsToRender={componentsToRender} />
```
Comment (12) [inline src/app/containers/InlineContainer/index.jsx]: So when this renders in the client, we get the following structure:
```
div.ItemConstrainedMedium
  p.Paragraph
    div.ItemConstrainedMedium
      span.InlineSpan
```
It's invalid HTML to nest a `div` in a ` p`, and as such, this error is thrown in the console
```
react-dom.development.js?61bb:506 Warning: validateDOMNesting(...): <div> cannot appear as a descendant of <p>.
```

File: src/app/containers/InlineContainer/index.jsx [hunk 2]
```diff
@@ -0,0 +1,26 @@
+import React, { useContext } from 'react';
+import { ServiceContext } from '../../contexts/ServiceContext';
+import Blocks from '../Blocks';
+import fragment from '../Fragment';
+import InlineLink from '../InlineLink';
+import InlineSpan from '../../components/InlineSpan';
+import inlineBlockPropTypes from '../../models/propTypes/inline';
```
Comment (12) [inline src/app/containers/InlineContainer/index.jsx]: Are these up to date?

https://github.com/bbc/simorgh/blob/26154fc29befe062bdace2e2e77532d7ad679df1/src/app/models/propTypes/inlineLink/index.js#L5-L14

Seeing this error in console

```
Failed prop type: The prop `isRequired` is marked as required in `InlineContainer`, but its value is `undefined`.
```

**Review by 12 — CHANGES_REQUESTED** [review]
I see an inline block rendering! 🎉 

However, I've got four console errors. 

```
checkPropTypes.js?a15c:20 Warning: Failed prop type: checker is not a function
    in ArticleMain (created by ArticleContainer)
    in ArticleContainer (created by DataContainer)
    in DataContainer (created by ErrorContainer)
    in ErrorContainer (created by LoadingContainer)
    in LoadingContainer (created by PageWrapperContainer)
    in PageWrapper (created by PageWrapperContainer)
    in PageWrapperContainer (created by WithContextsContainer)
    in RequestContextProvider (created by WithContextsContainer)
    in LoadableComponent (created by ServiceContextProvider)
    in ServiceContextProvider (created by WithContextsContainer)
    in DialContextProvider (created by WithContextsContainer)
    in ToggleContextProvider (created by WithContextsContainer)
    in WithContextsContainer (created by Context.Consumer)
    in Route (created by App)
    in Switch (created by App)
    in App (created by Context.Consumer)
    in withRouter(App) (created by ClientApp)
    in Router (created by BrowserRouter)
    in BrowserRouter (created by ClientApp)
    in ClientApp
```

```
react-dom.development.js?61bb:506 Warning: Expected server HTML to contain a matching <div> in <p>.
```

The others are posted below.

File: src/app/models/propTypes/inline/index.js [hunk 3]
```diff
@@ -1,11 +1,15 @@
-import { string, arrayOf, oneOf, shape } from 'prop-types';
+import { string, arrayOf, oneOfType, shape } from 'prop-types';
```
Comment (twinlensreflex) [inline src/app/models/propTypes/inline/index.js]: I changed this after reading this [blog post on oneOf vs oneOfType](https://jaketrent.com/post/react-oneof-vs-oneoftype/)

File: src/app/components/InlineSpan/index.jsx [hunk 4]
```diff
@@ -0,0 +1,23 @@
...
+import { C_SHADOW } from '@bbc/psammead-styles/colours';
+import { GEL_SPACING_TRPL } from '@bbc/gel-foundations/spacings';
+import { getBodyCopy } from '@bbc/gel-foundations/typography';
+import { scriptPropType } from '@bbc/gel-foundations/prop-types';
+import { getSansRegular } from '@bbc/psammead-styles/font-styles';
+
+const InlineSpan = styled.span`
+  ${props => (props.script ? getBodyCopy(props.script) : '')};
+  color: ${C_SHADOW};
+  ${({ service }) => getSansRegular(service)}
```
Comment (dr3) [inline src/app/components/InlineSpan/index.jsx]: Does this need this? Would it not be better to inherit the font styles from the parent, avoids the possibility of a mis-match?

Comment (dr3) [reply src/app/components/InlineSpan/index.jsx]: Actually bodyCopy, color are inherited too maybe?

Comment (jamesdonoh) [reply src/app/components/InlineSpan/index.jsx]: I've removed these lines from the PR in e099386 and can't see any visual difference so I guess not needed.

**Comment by jamesdonoh** [comment]
Additions/changes after handover:

- Fixed all outstanding propType warnings which seem to be mainly due to confusion about the propType helper functions (6b0605c)
- Removed unneeded styles following Drew's suggestion at https://github.com/bbc/simorgh/pull/2791#pullrequestreview-270622853 (e099386)

**Review by 12 — APPROVED** [review]
Thanks for picking this up. All LGTM 🙂

**Review by RayNjeri — APPROVED** [review]
All good!

**Review by dr3 — APPROVED** [review]

**Comment by dr3** [comment]
Need to update snaps :)

**Comment by jamesdonoh** [comment]
*sad trumpet*

**Comment by greenc05** [comment]
Has this had a developer accessibility swarm? Including text, a link and an external link with the lang attribute?

**Comment by HarveyPeachey** [comment]
> external link with the lang attribute?

@greenc05 Me and James just did a swarm for this issue using the [asset](http://localhost.bbc.com:7080/news/articles/cv7yn5n25rno) provided by eleanor. 

Here's the test HTML for the inline block support:
```
<span lang="fa" class="InlineSpan-sc-1w2gf2n-0 ffibHJ">روباه چه میگوید</span>

<span lang="fa" class="InlineSpan-sc-1w2gf2n-0 ffibHJ"><a href="https://google.com" aria-label="چیسربرگر, external" class="InlineLink-hkwpaz-0 idscKS">چیسربرگر</a></span>
```

Here's an overview of what we found on each screen reader:

**JAWS**
- In the description above the inline block and the inline block link which has english text with a Farsi word at the end, the screen reader reads out the Farsi word, but it reads out a different word every-time you navigate to it.

- The Inline block span also follows the same behaviour reading out a different word each time.

- The Inline block link also follows the same behaviour however it read out "Link/Lang Fa" before it reads out the random Farsi word.

**Dragon**
- Not done any tests as there are no new interactive controls added.

**ZoomText**
- Using mouse hover and speak it tools, none of the farsi words are being read out.

- App Reader test tool is not working as it's greyed out, so we skipped that last step of testing.

**Read&Write**
- Same behaviour as zoomtext with both Read by sentence and Read by web where it doesn’t read out the Farsi text

**iOS**
- In the English blocks it doesn’t say the Farsi text however in the Farsi inline block it reads out the Farsi word in a different voice, which we would assume is correct. Also in the inline block link it is the same as the inline block which is then followed by the voice over saying “external link”.

What we need is someone who can translate the Farsi being said by the screen reader, so we know that's what it should be saying.

**Comment by greenc05** [comment]
In this example we have english text (external) that we are saying is Farsi? Thoughts? ```<span lang="fa" class="InlineSpan-sc-1w2gf2n-0 ffibHJ"><a href="https://google.com" aria-label="چیسربرگر, external" class="InlineLink-hkwpaz-0 idscKS">چیسربرگر</a></span>```

For Jaws "it reads out a different word every-time you navigate to it." Do we think this needs some further investigation? It maybe because the machine doesn't have the correct speech synthesiser? https://support.freedomscientific.com/Downloads/Synthesizers

**Comment by HarveyPeachey** [comment]
> In this example we have english text (external) that we are saying is Farsi? Thoughts? `<span lang="fa" class="InlineSpan-sc-1w2gf2n-0 ffibHJ"><a href="https://google.com" aria-label="چیسربرگر, external" class="InlineLink-hkwpaz-0 idscKS">چیسربرگر</a></span>`
> 
> For Jaws "it reads out a different word every-time you navigate to it." Do we think this needs some further investigation? It maybe because the machine doesn't have the correct speech synthesiser? https://support.freedomscientific.com/Downloads/Synthesizers

So as discussed with both @jamesdonoh and @greenc05 I've gone through the accessibility tests again but instead of Farsi I've used Spanish fixture data which is in the most recent commit, this is to work around not having a Farsi voice synthesiser, as the Spanish one is built in. Here are my findings:

**JAWS IE11**

- Works correctly as it switches voice when it hits the Spanish inline block and and inline block link

**NVDA**

- Works correctly as it switches voice when it hits the Spanish inline block and and inline block link

**Dragon**

- Not done any tests as there are no new interactive controls added.

**ZoomText**

- Spanish Inline block and inline block link are read out but not in a different accent, it also doesn’t state that the link is a link

**Read&Write**

- Spanish Inline block and inline block link are read out but not in a different accent, it also doesn’t state that the link is a link.

**iOS**

- In the English blocks it does say the Spanish text and in the Spanish inline block it reads out the Spanish word in a different voice, which is correct. Also in the inline block link, it is the same as the inline block and says external link in Spanish.

**Comment by greenc05** [comment]
Looks to be as expected, thank you @HarveyPeachey

**Pull request merged**